File: java-context-os/src/main/java/com/owencli/contextos/pipeline/middleware/PolicyMiddleware.java
package com.owencli.contextos.pipeline.middleware;

import com.owencli.contextos.pipeline.PipelineContext;
import com.owencli.contextos.pipeline.PipelineMiddleware;
import com.owencli.contextos.policy.ContextPolicy;
import java.util.concurrent.CompletableFuture;

public class PolicyMiddleware implements PipelineMiddleware {
    private final ContextPolicy cp;
    public PolicyMiddleware(ContextPolicy c) { this.cp = c; }
    public String name() { return "policy"; }
    public int order() { return 200; }
    public CompletableFuture<Void> execute(PipelineContext ctx) {
        ctx.setPolicyDirective(cp.evaluate(ctx.taskSpec()));
        return CompletableFuture.completedFuture(null);
    }
}

File: context_os/pipeline/middlewares/policy_middleware.py
"""PolicyMiddleware — 策略评估。"""
from context_os.pipeline.context import PipelineContext
from context_os.pipeline.middleware import PipelineMiddleware
from context_os.policy import ContextPolicy

class PolicyMiddleware(PipelineMiddleware):
    def __init__(self, policy: ContextPolicy): self._cp = policy
    def name(self): return "policy"
    def order(self): return 200
    async def execute(self, ctx: PipelineContext) -> None:
        ctx.policy_directive = self._cp.evaluate(ctx.task_spec)
